muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/aside.tsx

```diff
@@ -31,165 +41,277 @@ class NavigationAside extends React.Component<
   NavigationProps,
   NavigationState
 > {
+  // These are required to make the types work in the context
```

Comment:
For readability in mind, many of this component render logic could be separated to their own render methods. Like separating students and notes view, and even NavigationTopic related section. Not mandatory, but think it.